src/Generator/TracdelightCOM.php

```diff
@@ -285,117 +320,101 @@ private function returnImagePath($variation, int $number)
      */
     private function getVariationAttributes($variation, KeyValue $settings):array
     {
-        $variationAttributes = [];
-
-        // Go through all the attributes
-        foreach($variation['data']['attributes'] as $variationAttribute)
+        if(!array_key_exists($variation['id'], $this->attributesCache))
         {
-            $attributeValueName = $this->attributeValueNameRepository->findOne($variationAttribute['valueId'], $settings->get('lang'));
+            unset($this->attributesCache);
 
-            if($attributeValueName instanceof AttributeValueName)
+            $variationAttributes = [];
+
+            // Go through all the attributes
+            foreach($variation['data']['attributes'] as $variationAttribute)
             {
-                // Check if the attribute is available for Tracdelight
-                if($attributeValueName->attributeValue->tracdelightValue)
+                $attributeValueName = $this->attributeValueNameRepository->findOne($variationAttribute['valueId'], $settings->get('lang'));
+
+                if($attributeValueName instanceof AttributeValueName)
                 {
-                    // Get the color and size attribute value
-                    if(($attributeValueName->attributeValue->attribute->backendName == 'Color' || $attributeValueName->attributeValue->attribute->backendName == 'Size')
-                        && !is_null($attributeValueName->attributeValue->tracdelightValue))
+                    // Check if the attribute is available for Tracdelight
+                    if($attributeValueName->attributeValue->tracdelightValue)
                     {
-                        $variationAttributes[strtolower($attributeValueName->attributeValue->attribute->backendName)] = $attributeValueName->attributeValue->tracdelightValue;
+                        // Get the color and size attribute value
+                        if(($attributeValueName->attributeValue->attribute->backendName == 'Color' || $attributeValueName->attributeValue->attribute->backendName == 'Size')
+                            && !is_null($attributeValueName->attributeValue->tracdelightValue))
```

Comment:
You have checked that value before

Reply:
Modification done and I've added also two more type of attributes that I needed. I've retested right now and it works well.